### PR TITLE
perf: push traces iff `SP1_DEBUG` is set in v2 compiler

### PR DESCRIPTION
**Pull request by tqn**

### CODE REVIEW

File: recursion/compiler/src/ir/builder.rs [hunk 1]
```diff
@@ -49,18 +50,11 @@ impl<T> TracedVec<T> {
...
-                self.traces.push(Some(Backtrace::new_unresolved()));
-            }
-            _ => {
-                self.traces.push(None);
-            }
-        };
+        if sp1_debug_mode() {
+            self.traces.push(Some(Backtrace::new_unresolved()));
+        } else {
+            self.traces.push(None);
```
Comment (tamirhemo) [inline recursion/compiler/src/ir/builder.rs:56]: Is it necessary to have this vector of `None`?

Comment (tqn) [reply recursion/compiler/src/ir/builder.rs:56]: I think the old runtime may panic because it will try to look up a trace by index in `traces`.

Comment (tamirhemo) [reply recursion/compiler/src/ir/builder.rs:56]: Oh I see now that this is the code that was here before. I think you can leave it as it for now.

**Review by tamirhemo — APPROVED** [review]
LGTM

**Pull request merged**